elastic/kibana
# [Guided onboarding] Implement quit guide functionality

This PR implements the "Exit setup guide" functionality in the guide dropdown panel. Once the user confirms to quit a guide, an API request is made to delete the SO that is storing the guide's state.

Fixes https://github.com/elastic/kibana/issues/139804

### How to test
1. Enable guided onboarding via your `kibana.dev.yml` file: `guidedOnboarding.ui: true`
2. Start up Kibana with the example plugins enabled: `yarn start --run-examples`
3. Navigate to `/app/guidedOnboardingExample`
4. Click the "Start search guide" button to trigger the guide
5. Click the "Quit setup guide" link at the button of the dropdown panel and verify modal displays.
6. Confirm clicking "Cancel" or the "x" icon closes the modal and maintains the guide state.
7. Confirm clicking "Quit guide" exits the guide. The Setup guide button should be disabled (behavior will change once https://github.com/elastic/kibana/issues/141129 is implemented). Also, if you navigate back to `/app/guidedOnboardingExample` the Search guide should say "Continue".


### Screenshots

<img width="555" alt="Screen Shot 2022-10-10 at 9 31 13 AM" src="https://user-images.githubusercontent.com/5226211/194878170-2a48fba0-ead8-470c-83b5-20aa11af2154.png">


// Loading state while API request is in progress
<img width="587" alt="Screen Shot 2022-10-05 at 9 04 00 AM" src="https://user-images.githubusercontent.com/5226211/194073565-beef1cfc-ada3-49bd-ae4c-0b9b6eb6e799.png">
